kernel-modules/support-packages/04-create-support-packages.sh

```diff
@@ -22,10 +22,24 @@ compress_files() (
     zip -r "${output_file}" .
 )
 
+use_downstream() {
+    IFS='.' read -ra version <<< "${1%-*}"
```

Comment:
There should be some input validation. Are we getting three integers?

Reply:
Why wouldn't we get 3 integers? The input this function gets is the driver version that has been changed to a `M.m.p` format for quite some time now.

Reply:
You could do something like

    if [[ ! $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
        echo "Error: Invalid version number format. Expected format: major.minor.patch"
        return 1
    fi
    
  Let's not make assumptions about what the format might be like in the future.